pandas/core/dimensioned.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+
+from pandas.core.base import (PandasObject)
+from pandas.util._decorators import cache_readonly
+from pandas import compat
+from pandas.core.common import is_null_slice
+
+
+class Dimensional(PandasObject):
+    """
```

Comment:
Sorry, I don't get it.
I looked at `DatetimeTZDtype`, but it has its own block type `DatetimeTZBlock`.
I looked at categorical data, it has its own `class Categorical(PandasObject)`. 
I looked at the PeriodDtype, but it is only used for indexes and I cannot instantiate a Series like `pd.Series([1,2,3], dtype="period[S]")`.

I'm sorry if I overlooked it, but if it is all there, I just don't see where it is.
I can create the Dtype easily enough, I just can't get it to integrate well with pd.Series.